core/invoker/src/main/scala/whisk/core/container/Container.scala

```diff
@@ -49,7 +49,7 @@ class Container(
     implicit var transid = originalId
 
     val id = Container.idCounter.next()
-    val name = containerName.getOrElse("anon")
+    val name = containerName.getOrElse(ContainerName.fromString("anon"))
```

Comment:
Good catch, but I think the fix should be `containerName.map(_.toString).getOrElse("anon")`. We shouldn't create an instance of the typed `ContainerName` from the gargabe value `anon`.